array-manipulation-2/src/chunk.ts

```diff
@@ -0,0 +1,24 @@
+/* exported chunk */
+function chunk(array: unknown[], size: number): unknown[] {
+  const chunkArray = [];
+  let chunk = [];
+  const mostChunks = Math.floor(array.length / size);
+  for (let i = 0; i < mostChunks * size; i += size) {
+    chunk = [];
+    for (let y = 0; y < size; y++) {
+      chunk.push(array[i + y]);
+    }
+    chunkArray.push(chunk);
+  }
+  const remaining = array.length % size;
+  console.log(remaining);
+  if (remaining !== 0) {
+    chunk = [];
+    for (let r = array.length - remaining; r < array.length; r++) {
+      chunk.push(array[r]);
+    }
+    chunkArray.push(chunk);
+  }
+  console.log(chunkArray);
+  return chunkArray;
+}
```

Comment:
I will approve but before you merge and delete branch, I would challenge you to solve this with only using a single for loop.

You will learn about Big O Notation and time efficiency in the future, but the tl;dr notes is that nesting `for` loops will make your program less efficient since you have to run a new loop during each iteration.

Reply:
Got it. I will work on that.